app/controllers/org_admin/templates_controller.rb

```diff
@@ -133,47 +135,36 @@ def create
     # -----------------------------------------------------
     def update
       template = Template.find(params[:id])
-      authorize template   # NOTE if non-authorized error is raised, it performs a redirect to root_path and no JSON output is generated
-
+      authorize template 
       begin
-        # Commenting this out for ticket #1476 which requests that we not version a template
-        # if its details change (only for phase/section/question changes). Leaving the line
-        # below in the event that we determine that some attributes (e.g. title) should trigger
-        # a new version
-        #template = Template.find_or_generate_version!(template)
-
+        template.assign_attributes(template_params)
         template.links = ActiveSupport::JSON.decode(params["template-links"]) if params["template-links"].present?
-        template.description = params["template-desc"]
+        if template.save!
+          render(status: :ok, json: { msg: success_message(template_type(template), _('saved'))})
+        else
+          # Note failed_update_error may return HTML tags (e.g. <br/>) and therefore the client should parse them accordingly
+          render(status: :bad_request, json: { msg: failed_update_error(template, template_type(template))})
+        end
       rescue ActiveSupport::JSON.parse_error
         render(status: :bad_request, json: { msg: _("Error parsing links for a #{template_type(template)}") }) and return
       rescue => e
         render(status: :forbidden, json: { msg: e.message }) and return
       end
-
-      if template.update_attributes(params[:template])
-        render(status: :ok, json: { msg: success_message(template_type(template), _('saved'))})
-      else
-        # Note failed_update_error may return HTML tags (e.g. <br/>) and therefore the client should parse them accordingly
-        render(status: :bad_request, json: { msg: failed_update_error(template, template_type(template))})
-      end
     end
     
     # DELETE /org_admin/templates/:id
     # -----------------------------------------------------
     def destroy
       template = Template.find(params[:id])
       authorize template
-      if template.plans.length <= 0
-        # Only allow the current version to be destroyed
-        if template.latest?
-          if template.destroy
+      versions = Template.includes(:plans).where(family_id: template.family_id)
```

Comment:
the action should destroy a template given an id therefore you only have to check whether or not that template has plans associated. If it has plans, you can't destroy it... I don't understand why do you want to destroy all its versions too when they have no plans...
If we really want that case, I'm more inclined to create another action like 'destroy_all_versions' to accommodate what you are trying to do here

Reply:
Where is that conversation tracked?